File: JavaScript/Ejercicio2.js
// Crear tipo definido / Clase
class Persona {
    constructor(nombre, apellido1, apellido2) {
        this.nombre = nombre;
        this.apellido1 = apellido1;
        this.apellido2 = apellido2;
    }
    
    toString() {
        return `${this.nombre} ${this.apellido1} ${this.apellido2}`;
    }
}

// Utilizarlo en un Array
let personas = [
    new Persona("Juan", "García", "López"),
    new Persona("María", "Rodríguez", "Martín"),
    new Persona("Carlos", "Hernández", "Pérez")
];

// Mostrar el array
console.log("Array de Personas:");
personas.forEach((persona, index) => {
    console.log(`[${index}]: ${persona.toString()}`);
});

File: C++/Ejercicio2.cpp
#include <iostream>
#include <string>
using namespace std;

// Crear tipo definido / Struct
struct Persona {
    string nombre, apellido1, apellido2;
    
    Persona(string n, string a1, string a2) {
        nombre = n;
        apellido1 = a1;
        apellido2 = a2;
    }
    
    string toString() {
        return nombre + " " + apellido1 + " " + apellido2;
    }
};

// Utilizarlo en un Array
int main() {
    Persona personas[3] = {
        Persona("Juan", "García", "López"),
        Persona("María", "Rodríguez", "Martín"),
        Persona("Carlos", "Hernández", "Pérez")
    };
    
    // Mostrar el array
    cout << "Array de Personas:" << endl;
    for (int i = 0; i < 3; i++) {
        cout << "[" << i << "]: " << personas[i].toString() << endl;
    }
    
    return 0;
}